packages/mui-material/src/Popper/Popper.tsx

```diff
@@ -0,0 +1,211 @@
+
+/**
+ *
+ * Demos:
+ *
+ * - [Autocomplete](https://mui.com/components/autocomplete/)
+ * - [Menus](https://mui.com/components/menus/)
+ * - [Popper](https://mui.com/components/popper/)
+ *
+ * API:
+ *
+ * - [Popper API](https://mui.com/api/popper/)
+ */
+const Popper = React.forwardRef(function Popper(
+  props: PopperProps,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  const theme = useTheme<{ direction?: Direction }>();
+  return <PopperUnstyled direction={theme?.direction} {...props} ref={ref} />;
+});
+
+function resolveAnchorEl(anchorEl: any) {
+  return typeof anchorEl === 'function' ? anchorEl() : anchorEl;
+}
+
+Popper.propTypes /* remove-proptypes */ = {
+  // ----------------------------- Warning --------------------------------
+  // | These PropTypes are generated from the TypeScript type definitions |
+  // |     To update them edit TypeScript types and run "yarn proptypes"  |
+  // ----------------------------------------------------------------------
+  /**
+   * An HTML element, [virtualElement](https://popper.js.org/docs/v2/virtual-elements/),
+   * or a function that returns either.
+   * It's used to set the position of the popper.
+   * The return value will passed as the reference object of the Popper instance.
+   */
+  anchorEl: chainPropTypes(
+    PropTypes.oneOfType([HTMLElementType, PropTypes.object, PropTypes.func]),
+    (props) => {
+      if (props.open) {
+        const resolvedAnchorEl = resolveAnchorEl(props.anchorEl);
+
+        if (resolvedAnchorEl && resolvedAnchorEl.nodeType === 1) {
+          const box = resolvedAnchorEl.getBoundingClientRect();
+
+          if (
+            process.env.NODE_ENV !== 'test' &&
+            box.top === 0 &&
+            box.left === 0 &&
+            box.right === 0 &&
+            box.bottom === 0
+          ) {
+            return new Error(
+              [
+                'MUI: The `anchorEl` prop provided to the component is invalid.',
+                'The anchor element should be part of the document layout.',
+                "Make sure the element is present in the document or that it's not display none.",
+              ].join('\n'),
+            );
+          }
+        } else if (
+          !resolvedAnchorEl ||
+          typeof resolvedAnchorEl.getBoundingClientRect !== 'function' ||
+          (resolvedAnchorEl.contextElement != null &&
+            resolvedAnchorEl.contextElement.nodeType !== 1)
+        ) {
+          return new Error(
+            [
+              'MUI: The `anchorEl` prop provided to the component is invalid.',
+              'It should be an HTML element instance or a virtualElement ',
+              '(https://popper.js.org/docs/v2/virtual-elements/).',
+            ].join('\n'),
+          );
+        }
+      }
+
+      return null;
+    },
+  ),
+  /**
+   * Popper render function or node.
+   */
+  children: PropTypes /* @typescript-to-proptypes-ignore */.oneOfType([
```

Comment:
Same as above and the ones below. I believe that any props with `@typescript-to-proptypes-ignore` or chainPropTypes are ignored by the TypeScript -> prop-types generation script.